transaction/signer/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Transaction signer types, used for communication with external signer
+//! implementations
```

Comment:
This description is rather sparse, can we try to add some description of what an external signer does, or maybe talk about an example?